lib/puppet/indirector/catalog/certless.rb

```diff
@@ -9,72 +9,31 @@ class Catalog
       class Certless < Puppet::Indirector::REST
         desc "Find certless catalogs over HTTP via REST."
 
-        # Override the REST indirector headers for the certless
-        # catalog indirector as the `puppet/pson` header throws of the
-        # request body and wraps it in a "value":{<request>} which
-        # causes a validation error on the v4/catalog endpoint
-        def headers
-          common_headers = {
-            "Content-Type" => 'text/json',
-            "Accept" => 'application/json',
-            Puppet::Network::HTTP::HEADER_PUPPET_VERSION => Puppet.version
-          }
-
-          add_accept_encoding(common_headers)
-        end
-
         def find(request)
-          uri = "/puppet/v4/catalog"
-
-          body = {
-            "certname": request.key,
-            "persistence": {
-              "facts": true, "catalog": true
-            },
-            "environment": request.environment.name.to_s,
-            "facts": {
-              "values": request.options[:transport_facts]
-            },
-            "trusted_facts": {
-              "values": request.options[:trusted_facts]
-            },
-            "transaction_uuid": request.options[:transaction_uuid],
-            "job_id": request.options[:job_id],
-            "options": {
-              "prefer_requested_environment": false,
-              "capture_logs": false
+          persistence = { facts: true, catalog: true }
+          opts = {
```

Comment:
The signature of the post_catalog4 method has been updated to accept effectively `certname, payload` where the `payload` is a hash with all the kwargs. instead of having some positional params and an "opts" hash, we now have some required kwargs. We should update our call signature to be `post_catalog4(certname, payload)`.